src/editor.c

```diff
@@ -1352,6 +1356,150 @@ static gint get_sci_line_code_end_position(ScintillaObject *sci, gint line)
 }
 
 
+/* returns the real indentation width, not the depth */
+static gint get_open_parenthesis_indent(ScintillaObject *sci, gint line, gint *line_matched)
+{
+	gint pos, pos_start = 0;
+	gint ret = -1, depth = 0;
+	gint lexer;
+
+	lexer = sci_get_lexer (sci);
+	pos = sci_get_line_end_position(sci, line) - 1;
+	/*if (one_line)
+		pos_start = sci_get_position_from_line(sci, line);*/
+	for (; pos >= pos_start && depth >= 0; pos--)
+	{
+		if (highlighting_is_code_style (lexer, sci_get_style_at (sci, pos)))
+		{
+			gchar ch = sci_get_char_at(sci, pos);
+
+			if (ch == ')')
+				depth++;
+			else if (ch == '(')
+				depth--;
+		}
+	}
+	if (depth < 0)
+	{
+		gint end_pos;
+		gint match_line;
+
+		end_pos = pos + 1;
+		match_line = sci_get_line_from_position(sci, end_pos);
+		if (line_matched)
+			*line_matched = match_line;
+		/* don't indent if it would use the full line size since it is unlikely
+		 * somebody wants it */
+		if (end_pos < sci_get_line_end_position (sci, match_line) - 1)
+		{
+			ret = sci_get_col_from_position(sci, end_pos + 1);
+			/* and then keep the smallest indent of a line between match and current,
+			 * not to restore a user reduced indent (e.g. for base_indent+1 style) */
+			for (++match_line; match_line < line; ++match_line)
```

Comment:
Well, I would have said the first is right. sure the user undented `bar(` likely to avoid the line getting too long, but I wouldn't have expected or wanted the next parameter to align with that, at least not automagically.